module/zfs/dataset_kstats.c

```diff
@@ -198,6 +198,19 @@ dataset_kstats_destroy(dataset_kstats_t *dk)
 	zil_sums_fini(&dk->dk_zil_sums);
 }
 
+void
+dataset_kstats_rename(dataset_kstats_t *dk, const char *name)
+{
+	dataset_kstat_values_t *dkv = dk->dk_kstats->ks_data;
+	char *ds_name;
+
+	ds_name = KSTAT_NAMED_STR_PTR(&dkv->dkv_ds_name);
+	ASSERT3S(ds_name, !=, NULL);
+	ASSERT3S(KSTAT_NAMED_STR_BUFLEN(&dkv->dkv_ds_name), ==,
+	    ZFS_MAX_DATASET_NAME_LEN);
+	(void) strlcpy(ds_name, name, ZFS_MAX_DATASET_NAME_LEN);
```

Comment:
Considering dataset_kstats_destroy() uses KSTAT_NAMED_STR_BUFLEN(), I would do it here also:
```suggestion
	(void) strlcpy(ds_name, name, KSTAT_NAMED_STR_BUFLEN(&dkv->dkv_ds_name));
```

Reply:
done